vector/src/main/java/im/vector/app/features/location/live/map/UserLiveLocationViewStateMapper.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location.live.map
+
+import im.vector.app.features.home.room.detail.timeline.helper.LocationPinProvider
+import im.vector.app.features.location.toLocationData
+import kotlinx.coroutines.suspendCancellableCoroutine
+import org.matrix.android.sdk.api.session.room.model.livelocation.LiveLocationShareAggregatedSummary
+import javax.inject.Inject
+
+class UserLiveLocationViewStateMapper @Inject constructor(
+        private val locationPinProvider: LocationPinProvider,
+) {
+
+    suspend fun map(liveLocationShareAggregatedSummary: LiveLocationShareAggregatedSummary) =
+            suspendCancellableCoroutine<UserLiveLocationViewState?> { continuation ->
+                val userId = liveLocationShareAggregatedSummary.userId
+                val locationData = liveLocationShareAggregatedSummary.lastLocationDataContent
+                        ?.getBestLocationInfo()
+                        ?.geoUri
+                        .toLocationData()
+
+                when {
+                    userId.isNullOrEmpty() || locationData == null -> continuation.resume(null) {
```

Comment:
do we need to use the `resume(value, callback)` overload of `resume`? assuming they're the same, it looks like we make use of `resume(value)` and avoid need the empty callback with comment

Reply:
I didn't found the `resume(value)` signature method, it seems the callback is mandatory...

Reply:
we're using the non callback version here  https://github.com/vector-im/element-android/blob/955f366690def07fa90877cb49224ca7cbab0f99/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/network/RetrofitExtensions.kt#L45

perhaps it's a different import :thinking: